src/mlagility/api/cloud.py

```diff
@@ -365,6 +396,82 @@ def execute_gpu_remotely(
     sys.stdout = sys.stdout.terminal
 
 
+def execute_gpu_locally(
+    state: build.State, log_execute_path: str, iterations: int
+) -> None:
+    """
+    Execute Model on the local GPU
+    """
+
+    # Redirect all stdout to log_file
+    sys.stdout = build.Logger(log_execute_path)
```

Comment:
It feels weird to me that we're completely silent during the benchmark running. It would be nice to at least display a `tqdm` or something.

Reply:
There are multiple pieces in this flow such as network transfer, conda env creating/ docker setup and transferring files back. tqdm might not be so reliable in this case. We could do `Benchmarking in progress ...` with the blinker, as we have for the groqflow build process.

Reply:
This comment is about local execution, so there should be no network transfer. The tqdm can just wrap the 100 iterations of the benchmark

Reply:
Created https://github.com/groq/mlagility/issues/100